### PR TITLE
LibJS: Do not print large arrays

### PR DESCRIPTION
**Pull request by DasBasti**
Printing the whole array causes wpt [console/console-log-large-array.any.html](https://wpt.fyi/results/console/console-log-large-array.any.html?product=ladybird) to crash. 
This limits logged arrays to 100 elements and truncates the rest with ...
The runtime of the test on my machine went from 25 seconds down to 7.

The 100 is an arbitrary selected number so if you have a better solution I am happy to change it.

### CODE REVIEW

**Comment by AtkinsSJ** [comment]
> Also, printing a large array probably shouldn't crash, have you looked into the cause of it?

The test logs an array with 10 million elements. (Well, actually 2 arrays with 10 million elements each.) Some kind of truncation is necessary. :sweat_smile: Firefox and Chrome both divide the output up into expandable chunks so you still can theoretically view each element, but it doesn't try to dump everything out at once.

image

That seems like the best option but it's also not possible without implementing some kind of lazy-loading. IMO, this is a reasonable solution for now.

File: Userland/Libraries/LibJS/Print.cpp [hunk 1]
```diff
@@ -165,6 +165,7 @@ ErrorOr<void> print_array(JS::PrintContext& print_context, JS::Array const& arra
 {
     TRY(js_out(print_context, "["));
     bool first = true;
+    size_t cnt = 0;
```
Comment (AtkinsSJ) [inline Userland/Libraries/LibJS/Print.cpp:168]: We like to use descriptive variable names, something like `printed_count` would be better. (See [CodingStyle.md](https://github.com/LadybirdBrowser/ladybird/blob/master/Documentation/CodingStyle.md))

File: Userland/Libraries/LibJS/Print.cpp [hunk 2]
```diff
@@ -472,10 +477,15 @@ ErrorOr<void> print_typed_array(JS::PrintContext& print_context, JS::TypedArrayB
         TRY(js_out(print_context, "[ "));                                                \
         auto& typed_array = static_cast<JS::ClassName const&>(typed_array_base);         \
         auto data = typed_array.data();                                                  \
+        size_t cnt = 0;                                                                  \
```
Comment (AtkinsSJ) [inline Userland/Libraries/LibJS/Print.cpp:480]: ...same here.

**Comment by Hendiadyoin1** [comment]
Note for the future:
Printing, and lazy printing are observable on some objects
See #1112

**Review by AtkinsSJ — CHANGES_REQUESTED** [review]
Not crashing is good!

As circl mentioned, there's a typo in the commit title. But also, you seem to have stumbled across a loophole in our commit linter. :sweat_smile: Lines in the commit description should be wrapped at 72 characters. There's an exception for URLs, so I think the `console/console-log-large-array.any.html` part tricked it!

**Review by AtkinsSJ — APPROVED** [review]
Looks good to me! Thanks :^)

**Pull request merged**